crates/engine/tree/src/tree/mod.rs

```diff
@@ -1387,10 +1387,35 @@ where
                             self.canonical_in_memory_state.set_pending_block(block.clone());
                         }
 
+                        let insert_tree_start = Instant::now();
                         self.state.tree_state.insert_executed(block.clone());
+                        let insert_tree_elapsed = insert_tree_start.elapsed();
                         self.metrics.engine.inserted_already_executed_blocks.increment(1);
+                        let elapsed = now.elapsed();
+                        
+                        // X Layer: Update timing metrics for InsertExecutedBlock path
+                        use reth_node_metrics::block_timing::{get_block_timing, store_block_timing};
```

Comment:
The code is redundant. You can extract it into a helper function to reduce duplication and improve maintainability.